apps/champions/test/battle_test.exs

```diff
@@ -680,6 +680,158 @@ defmodule Champions.Test.BattleTest do
       assert "team_1" ==
                Champions.Battle.Simulator.run_battle([unit], [target_dummy_5], maximum_steps: maximum_steps).result
     end
+
+    test "Self Heal" do
+      maximum_steps = 5
+
+      # Create a character with a basic skill that will heal 5 damage to itself
+      heal_basic_skill_params =
+        TestUtils.build_skill(%{
+          name: "Heal Self skill",
+          mechanics: [
+            %{
+              trigger_delay: 0,
+              apply_effects_to:
+                TestUtils.build_apply_effects_to_mechanic(%{
+                  effects: [
+                    TestUtils.build_effect(%{
+                      executions: [
+                        %{
+                          type: "Heal",
+                          attack_ratio: 1,
+                          energy_recharge: 0
+                        }
+                      ]
+                    })
+                  ],
+                  targeting_strategy: %{
+                    type: "self",
+                    target_allies: false
+                  }
+                })
+            }
+          ],
+          cooldown: maximum_steps * @miliseconds_per_step - 1
+        })
+
+      {:ok, heal_character} =
+        TestUtils.build_character(%{
+          name: "Self Heal Character",
+          basic_skill: heal_basic_skill_params,
+          ultimate_skill: TestUtils.build_skill(%{name: "Self Heal Empty Skill"}),
+          base_attack: 5,
+          base_health: 10
+        })
+        |> Characters.insert_character()
+
+      {:ok, heal_unit} = TestUtils.build_unit(%{character_id: heal_character.id}) |> Units.insert_unit()
+      {:ok, heal_unit} = Units.get_unit(heal_unit.id)
+
+      deal_damage_basic_skill_params =
+        TestUtils.build_skill(%{
+          name: "DealDamage Skill for Self Test",
+          mechanics: [
+            %{
+              trigger_delay: 0,
```

Comment:
```suggestion
```

Reply:
Since there isn't a build_mechanic util, the trigger_delay field must be specified to avoid errors.